### PR TITLE
Hotfix for http message headers

### PR DESCRIPTION
**Pull request by slinkydeveloper**
Signed-off-by: Francesco Guardiani <francescoguard@gmail.com>

### CODE REVIEW

File: v2/protocol/http/message.go [hunk 1]
```diff
@@ -130,13 +130,21 @@ func (m *Message) ReadBinary(ctx context.Context, encoder binding.BinaryWriter)
 func (m *Message) GetAttribute(k spec.Kind) (spec.Attribute, interface{}) {
 	attr := m.version.AttributeFromKind(k)
 	if attr != nil {
-		return attr, m.Header[attributeHeadersMapping[attr.Name()]]
```
Comment (n3wscott) [inline v2/protocol/http/message.go:133]: use `m.Header.Get(attr.Name())`

File: v2/protocol/http/message.go [hunk 2]
```diff
@@ -130,13 +130,21 @@ func (m *Message) ReadBinary(ctx context.Context, encoder binding.BinaryWriter)
...
+			return attr, h[0]
+		}
+		return attr, nil
 	}
 	return nil, nil
 }
 
 func (m *Message) GetExtension(name string) interface{} {
-	return m.Header[extNameToHeaderName(name)]
+	h := m.Header[extNameToHeaderName(name)]
```
Comment (n3wscott) [inline v2/protocol/http/message.go:143]: use `m.Header.Get(extNameToHeaderName(name))`

Comment (slinkydeveloper) [reply v2/protocol/http/message.go:143]: I specifically avoided that b/c of avoiding additional string allocation (https://github.com/cloudevents/sdk-go/pull/444). This solution is equivalent

**Pull request merged**